### PR TITLE
tests: Add unit tests for InitCmd

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

#### Summary

* Starts e2e tests for the repository CLI. 
* Adds documentation for the InitCmd

This is incremental, I will be adding unit tests for all the command but it requires some refactoring. End goal is to include a fully signed repository and cosign initialize within these tests.

#### Ticket Link

Part of https://github.com/sigstore/root-signing/issues/237

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by asraa** [comment]
cc @znewman01 if you want to start reviewing PRs there's going to be a ton for each of the CLI commands

File: tests/e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+// Create fake key signer in testDirectory. Returns file reference to signer.
+func createTestSigner(t *testing.T) string {
+	keys, err := cosign.GenerateKeyPair(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	temp := t.TempDir()
+	path := path.Join(temp, "cosign.key")
```
Comment (Hayden-IO) [inline tests/e2e_test.go]: nit: A different name besides cosign, since this is used for the snapshot/timestamp?

File: tests/e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/root-signing/cmd/tuf/app"
+	vapp "github.com/sigstore/root-signing/cmd/verify/app"
+
+	"github.com/theupdateframework/go-tuf"
+	"github.com/theupdateframework/go-tuf/data"
+)
+
+// Create test HSM keys.
```
Comment (Hayden-IO) [inline tests/e2e_test.go]: nit: This is a single key, correct?

I wonder if it's worth creating multiple keys to mirror production

**Review by Hayden-IO — DISMISSED** [review]

File: tests/e2e_test.go (hunk 2 above)
Comment (asraa) [reply tests/e2e_test.go]: It is just a single key.
 I mean to add a TODO -- Init doesn't require signing with anything, and there's only one test_data key in there. 

Adding a TODO to create keys programmatically, and also to test >1 threshold.

**Review by Hayden-IO — APPROVED** [review]

**Pull request merged**